### PR TITLE
Extract delayed transitions

### PR DESCRIPTION
**Pull request by farskid**

### CODE REVIEW

File: new-packages/ts-project/__tests__/transitions.test.ts [hunk 1]
```diff
@@ -3034,3 +3034,392 @@ test('should extract transition description (multi-line)', async () => {
...
+          "data": {
+            "context": {},
+          },
+          "edges": {
+            "edge-0": {
+              "data": {
+                "actions": [],
+                "description": undefined,
+                "eventTypeData": {
+                  "delay": "100",
```
Comment (Andarist) [inline new-packages/ts-project/__tests__/transitions.test.ts:3073]: I wonder - does Studio try to auto-detect if this is a number or a string?

File: new-packages/ts-project/__tests__/transitions.test.ts [hunk 2]
```diff
@@ -3034,3 +3034,392 @@ test('should extract transition description (multi-line)', async () => {
...
+    'tsconfig.json': JSON.stringify({}),
+    'index.ts': ts`
+      import { createMachine } from "xstate";
+
+      createMachine({
+        initial: "foo",
+        states: {
+          foo: {
+            after: {
+              "named delay": "bar",
```
Comment (Andarist) [inline new-packages/ts-project/__tests__/transitions.test.ts:3160]: I'd replace this with:
```suggestion
              myDelay: "bar",
```

and add an extra test case with a property like the one u have here

File: new-packages/ts-project/__tests__/transitions.test.ts (hunk 1 above)
Comment (farskid) [reply new-packages/ts-project/__tests__/transitions.test.ts:3073]: It does but for formatting and displaying purposes. The delay that's stored in digraph definition is always a string.

File: new-packages/ts-project/__tests__/transitions.test.ts (hunk 2 above)
Comment (farskid) [reply new-packages/ts-project/__tests__/transitions.test.ts:3160]: I don't understand. Are you suggesting adding a third test for an identifier delay?

File: new-packages/ts-project/__tests__/transitions.test.ts [hunk 3]
```diff
@@ -3838,3 +3838,504 @@ test('should raise error when transition.meta contains any value other than a pl
...
+            },
+          },
+          "root": "state-0",
+        },
+        [],
+      ],
+    ]
+  `);
+});
+test('should extract after transition (string delay)', async () => {
```
Comment (Andarist) [inline new-packages/ts-project/__tests__/transitions.test.ts:4065]:
```suggestion
test('should extract after transition (string with whitespace delay)', async () => {
```

**Review by Andarist — APPROVED** [review]

File: new-packages/ts-project/__tests__/transitions.test.ts (hunk 2 above)
Comment (Andarist) [reply new-packages/ts-project/__tests__/transitions.test.ts]: ye, basically a test for the simplest test case (identifier, no whitespace) and a test for the more specific situation (string with whitespace)

File: new-packages/ts-project/__tests__/utils.ts [hunk 4]
```diff
@@ -98,7 +98,7 @@ async function createTestProgram(
     );
   }
 
-  const host = ts.createCompilerHost(options);
+  const host = ts.createCompilerHost(options, true);
```
[the remark this answers is not in the document]
Comment (farskid) [reply new-packages/ts-project/__tests__/utils.ts:101]: Let's revert it whenever later. Easy to do at any point.

**Pull request merged**